### DISCUSSION TITLE
Directory import is not supported?

### DISCUSSION THREAD

**Discussion by jtoon86** [primefaces/community#238] [PrimeReact] [answered]
I am learning Remix with PrimeReact and when I load the modules in the documented way, ie:

import { Button } from "primereact/button"

my server comes back with: Error: Directory import '.\node_modules\primereact\button' is not supported resolving ES modules imported from C:\Users\Laurie\js\remix-ts\build\index.js

I can use the recommended import to make it work:

import { Button } from "primereact/button/button.cjs.js"

but then  I get a typescript declaration error. 

I feel like I am missing something obvious so hopefully someone can point me in he right direction. Thanks!

If needed:
PrimeReact v10.0.3
Remix v2.0.1
Node v18.18.0

**Comment by habubey** [comment]
Hey there, is it possible to send us a reproducer?

**Comment by jtoon86** [comment, marked as answer]
I did some additional testing/searching and it seemed specific to Remix v2.  The changes that needed to be made to allow for standard primereact imports was as follows:

- In package.json - remove "type": "module" 
- Rename remix.config.js to remix.config.mjs
- Inside remix.config.mjs add the config option:  serverModuleFormat: "cjs"

Hopefully this helps if someone else trying to get this combination working comes to this discussion forum. :)
